File: packages/test-cases/cases/bind/sanity/output/wasm-as/Query/wrapped.ts
import {
  queryMethod,
  objectMethod
} from "../../index";
import {
  deserializequeryMethodArgs,
  serializequeryMethodResult,
  deserializeobjectMethodArgs,
  serializeobjectMethodResult
} from "./serialization";

export function queryMethodWrapped(argsBuf: ArrayBuffer): ArrayBuffer {
  const args = deserializequeryMethodArgs(argsBuf);
  const result = queryMethod({
    str: args.str,
    optStr: args.optStr,
    en: args.en,
    optEnum: args.optEnum,
    enumArray: args.enumArray,
    optEnumArray: args.optEnumArray
  });
  return serializequeryMethodResult(result);
}

export function objectMethodWrapped(argsBuf: ArrayBuffer): ArrayBuffer {
  const args = deserializeobjectMethodArgs(argsBuf);
  const result = objectMethod({
    object: args.object,
    optObject: args.optObject,
    objectArray: args.objectArray,
    optObjectArray: args.optObjectArray
  });
  return serializeobjectMethodResult(result);
}

File: packages/test-cases/cases/bind/sanity/output/wasm-rs/query/src/query/wrapped.rs
use crate::{
    deserialize_object_method_args, deserialize_query_method_args, serialize_object_method_result,
    serialize_query_method_result,
};
use crate::{object_method, query_method, InputObjectMethod, InputQueryMethod};

pub fn query_method_wrapped(args_buf: &[u8]) -> Vec<u8> {
    let args = deserialize_query_method_args(args_buf).expect("Failed to deserialize buffer");
    let result = query_method(InputQueryMethod {
        str: args.str,
        opt_str: args.opt_str,
        en: args.en,
        opt_enum: args.opt_enum,
        enum_array: args.enum_array,
        opt_enum_array: args.opt_enum_array,
    });
    serialize_query_method_result(result)
}

pub fn object_method_wrapped(args_buf: &[u8]) -> Vec<u8> {
    let args = deserialize_object_method_args(args_buf).expect("Failed to deserialize buffer");
    let result = object_method(InputObjectMethod {
        object: args.object,
        opt_object: args.opt_object,
        object_array: args.object_array,
        opt_object_array: args.opt_object_array,
    });
    serialize_object_method_result(result)
}
